docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,291 @@
+field containing the list of supported platforms, and will trigger tests on
+those platforms. Users will then be able to see the results of those tests on
+each of the platforms defined, giving them quality assurance and coverage
+there.
+
+The exact format of the metadata file is still to be defined, but a possibility
+is to define a JSON schema that will be provided on this new repository.
+
+Maintainability
+===============
+
+As described above, each library must follow the new standard and keep that
+same level for future versions. Because of that, new PRs must be approved by at
+least one MAINTAINER and pass the basic checks pipelines that will be provided
+as part of the autils repository.
+
+Besides the automatic checks, each MAINTAINER is responsible for keeping the
+consistency of that library with the overall project.
+
+Release
+=======
+
+As a new and standalone repository, there will not be any type of release
+version sync between Avocado and autils.  Avocado may release, say, version
+100.0, while autils won’t have a release. Because of that, autils will be
+versioned at a different pace than Avocado.
+
+Every week an automatic job will be executed to determine if there is a
+potential need for a new release:
+
+* If the number of commits since last release has reached a threshold; AND
+* There is no open discussion with the tag “release-decision”:
+  * Creates a new discussion (with the "release-decision" tag) to run a poll
+    between all the MAINTAINERS, so they will have the opportunity to thumbs
+    up if there is a need for a new release;
+  * The link to the new discussion should be sent to all MAINTAINERS;
+  * If the decision is positive, a round-robin selected maintainer should be
+    responsible for coordinating the release it self;
+  * And the discussion should be closed
+* If there is any existing open discussion, the bot job should comment on the
+  same discussion with the updated list of commits and ping again the
+  MAINTAINERS, give them the opportunity to thumbs up based on the new
+  status;
+
+Manual release process could also be started if a MAINTAINER think it is a good
+timing (i.e: Security bug or critical bug it was fixed).
+
+Backwards Compatibility
+***********************
+
+The current libraries are not going to be removed right away. A transition
+period is expected for migrating each library into this new standard, and
+because of that libraries in transition should be available, receive bug fixes
+for at least one LTS cycle.
+
+Each library in our current code base, should receive a status flag, informing
+users what their current status is, according on the following table:
+
+1. maintainer-wanted
+
+  The library is using the legacy mode, and it is not yet converted to the
+  autils repository. This should be the status for all avocado.utils.*
+  (including Avocado-VT utils) libraries upon the creation of the autils
```

Comment:
What scope of utilities is in autils as expected? Could all host related utilities be candidates? 

Are virtualization(VM) related utilities put into autils or still kept in avocado-vt? 
 
avocado-vt utils could be divided into below parts:
1. duplicated with avocado.utils.* in most extents which are not related to VM (virtualization)
2. duplicated with avocado.utils.* in some extents which are related to VM
3. consume avocado.utils.* 
4. isolated from avocado.utils.* 

For 1, move to autils
For 2, keep in avocado-vt
For 3 and 4, keep in avocado-vt

What do you think of? @beraldoleal @clebergnu

Reply:
@dzhengfy What if we have nonvirtualization utils like (1) that consume avocado.utils.* like (3)? What I mean is that I don't think your separation info four category is ideal as some of the types are not mutually exclusive.

Reply:
Also note that Avocado is already starting to duplicate some virtualization related functionality in the `vmimage` utility there so virtualization might actually end up needed beyond Avocado VT for spawners and other more general implementations. So I am not sure if "virtualization" will really make sense to stay in Avocado VT at least when it comes to reusable virtualization scope. Alternatively we might make sure that Avocado is always used in conjunction with Avocado VT as its main virtualization plugin.

Reply:
> @dzhengfy What if we have nonvirtualization utils like (1) that consume avocado.utils.* like (3)? What I mean is that I don't think your separation info four category is ideal as some of the types are not mutually exclusive.

Yeah, those categories might be not very accurate as I know. The main purpose is to help clarify what the autils' scope in deed. When we are very clear about the autil's scope, then we can make the categories more accurate and decide how avocado-vt utils will go. Thanks for your caring about.

Reply:
I agree with @pevogam here. I don't see any reason for keeping virtualization libraries outside autils project. At first, we could migrate libraries that are duplicated no matter the subject (except for the cases described at the utility onboarding section).

At the same section we describe the "scope" there. It is really hard to limit the scope here at this document. Things can change quickly, so I think it might be easier for us to just review proposals to see if that makes sense or not. We may have particularities for each case.